cg/cli/archive.py

```diff
@@ -64,3 +64,16 @@ def update_job_statuses(context: CGConfig):
         data_flow_config=context.data_flow,
     )
     spring_archive_api.update_statuses_for_ongoing_tasks()
+
+
+@archive.command("delete-file")
```

Comment:
I'm wondering if it would be useful to add a help message describing the steps taken (the ones you described in the PR description) so that the user is sure that all the checks are done internally. I' saying this because one sometimes is reluctant to run a command named `delete` before making sure it does some checks.

Reply:
Another cool feature of a delete command is to show a confirmation statement in which the user can change their mind when they see the path of the file to be deleted. With its comes also the option `--yes` to skip the confirmations

Reply:
Might be a good thing to add! In this case the only input provided is the path to the file but something similar might be valuable.

Reply:
Added a longer docstring (which is printed when using --help: 
<img width="606" alt="image" src="https://github.com/Clinical-Genomics/cg/assets/40887124/5f5bb59f-30af-4825-b294-06b11b9716ac">
) and added a dry run option. Is that sufficient? @diitaz93

Reply:
With dry-run set:

```
1d [isak.ohlsson@hasta:~] [S_base] 22s $ cg archive delete-file -d isaks_test_bundle/2023-12-06/isaks_test_spring_file.spring
Found archive location karolinska_bucket
Would have deleted file isaks_test_bundle/2023-12-06/isaks_test_spring_file.spring from karolinska_bucket.
```

Reply:
that is great!